Update before_filter to before_action

scottmatthewman opened the pull request:
To avoid Rails 5.1+ compatibility issues, rename the `before_filter` method calls to `before_action`.

Also do a little tidying up of the code (yes, yes, I know this should really belong in a different PR)

Commit:
Do not include Gemfile.lock in gems

See https://yehudakatz.com/2010/12/16/clarifying-the-roles-of-the-gemspec-and-gemfile/

Commit:
Use secure https link to rollout gem

Commit:
Switch homepage reference to local fork

Commit:
Rename deprecated before_filter to `before_action`

kieranogrady commented on rollout_ui.gemspec:
```diff
@@ -9,7 +9,7 @@ Gem::Specification.new do |gem|
   gem.email         = ["jrallison@gmail.com"]
```
Should we change email to `developers@trx.tv`?

scottmatthewman replied:
:+1:

scottmatthewman commented:
:+1:
On 7 Aug 2019, 14:28 +0100, kieranogrady <notifications@github.com>, wrote:

@kieranogrady commented on this pull request.

________________________________

In rollout_ui.gemspec<https://github.com/rightsxchange/rollout_ui/pull/1#discussion_r311551914>:

> @@ -9,7 +9,7 @@ Gem::Specification.new do |gem|
   gem.email         = ["jrallison@gmail.com"]


Should we change email to developers@trx.tv?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/rightsxchange/rollout_ui/pull/1?email_source=notifications&email_token=AAAEOUBW53MOGR7RJNLFGLDQDLEXPA5CNFSM4II4OSA2YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCA24PVA#pullrequestreview-271960020>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AAAEOUH5PIYUYVRYPFJVD2LQDLEXPANCNFSM4II4OSAQ>.

Commit:
Update gemspec with TRX contact info

In code review it was suggested that (correctly) we should include
contact information for our forked version.

I'm not sure what the etiquette is but it feels like wiping out the
original author's credit would be harsh, even if they're no longer
maintaining it.

Have also included a link to the original in the description. I think
mentioning the fork which we forked from may be going a bit too far.
